### ISSUE TITLE
Better handling for monolithic kernels

### ISSUE THREAD

**Issue by zackw** [dracut-ng/dracut-ng#691]
dracut can generate an initrd for a monolithic kernel (`CONFIG_MODULES=n`) but not a well-optimized one.  Problems I have noticed so far include:

* --host-only mode cannot be used; it sees that /proc/modules does not exist, throws up its hands, and refuses to continue
* the dracut modules dracut-systemd, systemd-initrd, systemd-udevd, and systemd-sysctl, all of which *should* work just fine, cannot be used because they have a transitive hard dependency on systemd-modules-load (which should be unnecessary with a monolithic kernel)
* the initrd includes kmod and all its aliases even though these are never useful with a monolithic kernel (the only reason they are even available to be copied into the initrd is that depmod *is* still useful).

I expect I will find more problems as I continue testing; I will follow up to this issue as necessary.

Ideally, when dracut is invoked on a system with a monolithic kernel, the only consequence would be that everything related to kernel modules is left out of the initrd; all the other components should work just the same.

**Comment by jozzsi** [comment]
> * the dracut modules dracut-systemd, systemd-initrd, systemd-udevd, and systemd-sysctl, all of which _should_ work just fine, cannot be used because they have a transitive hard dependency on systemd-modules-load (which should be unnecessary with a monolithic kernel)

Even after https://github.com/dracut-ng/dracut-ng/pull/637 ?

**Comment by Anoncheg1** [comment]
I am trying to install with btrfs submodules. CONFIG_MODULES=n
version 103-r4 
Without success now. Please fix.

**Comment by Hawk777** [comment]
@mutedbytes did you see [my workaround](https://github.com/dracut-ng/dracut-ng/issues/691#issuecomment-2867786880)? It’s not ideal but it does the job.

**Comment by mutedbytes** [comment]
@Hawk777 I did, but I believe this bug should still be fixed and generally do not want to rely on a workaround, especially because I am not calling dracut myself most of the time and relying on kernel-install to call it as part of kernel build/install. Also I do not want to carry a permament empty dummy file and env variable of which may or may not become a problem in the future and that I may forget the meaning of or intent of in the future.

**Comment by mutedbytes** [comment]
@Hawk777 Yes thank you for sharing your finding, it is indeed useful to function for now until a fix (very hopefully) comes.

**Comment by mutedbytes** [comment]
@jozzsi That does appear to work fine. Thank you for regarding this!

**Comment by mutedbytes** [comment]
In any case, I believe with the patches as @jozzsi has proposed in #1821 and #1823 which are slightly different than the patch commented above, as long as you intentionally use both `--hostonly` and `--no-kernel` then there will be no check or reliance on `/proc/modules` at all.
